### PR TITLE
Run Prettier on All The Things

### LINKED ISSUE #2875: formatting rules for SCSS files
Context: https://github.com/desktop/desktop/pull/2848

> p.s on saving, the spacing between lines 3/4 get's removed. Is that related to my vscode editor?

We're already prettier-ifying things, but it seems like VSCode is now eager to format these. An `.editorconfig` file at the root would work for all editors that respect this.

For example:

```
# top-most EditorConfig file
root = true

# Unix-style newlines with a newline ending every file
[*]
end_of_line = lf
insert_final_newline = true

# 4 space indentation
[*.scss]
indent_style = space
indent_size = 4
```

There are some other VSCode specific formatters (and probably things for Atom too) but I'd rather make this generic and let people work wherever they want.

### PR DESCRIPTION
**Pull request by j-f1**
Fixes #2875

- [x] Extract options to `.prettierrc`
- ~❔ Lint the JS/TS in Markdown (`eslint-plugin-markdown`) (` ```ts ` blocks aren’t supported yet) (blocked on https://github.com/eslint/eslint-plugin-markdown/issues/82)~
- [x] Prettier the SCSS
- [x] Prettier the Markdown ([It works 💎](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAVOBnGSA6UAGhMGeAZgK5RgwCW0ABABJwA2LEA6hAE4sAmACmABzbnDi0ow+gF56OEAAtW7BQBp6o8fD6z6AcgVdefBfo0YaLBDD2kAhiwxwN0ALIRyzgPIA3ONxqAL4AlPTAeHj09DSkAgCEWhI0UiHAYjDk3FD0UORsQQDckTnR0QD05fSo3gAi3kj0tdD6tl5w9Nz2UHwQALYxOWI9AVH01rZ5A3Ju9jCKAHSk7Dz0ArPzC109-QJhAFT0AIwAogDUAOwhCzAQAMow3CnCe1twAA4s9mBwAuU4OAWAL4Z3KNGEGgUChCJWiGSyOQAPHwaL56GAvhgMAA5ex9OAyfTMNicHj8fT0WgwawyYD4ACanno9jE9F8NEst24uXIfQARgF6AASYA8gZBfBBejuTw+fzcWky9p+AJBAB8sLKiKw3GgwjVoqSkmECwwLBoPwE9AADBojvRrrcAKrvd4BADC9mce3oZ00YmSUlN5stR0dEAAMhAAO4er2-MJBRHlHV6jWlaIiAPGhYIPgYDg0eYCdTQ+gAfnkICr9Ea2ve3XoWAAnjTgMBIOxuI1DDgtM39EF1eoq8mMA2oGr6EExtFEXA+umcDBDdm4LoZ1Bl8mF+myuE1pZrLAYRmypWFECQLOyo0FPEFJub8mUb41cUt1BN4R8CA1CAIHeWhoAwZBQBZXVowABRZBBQJQexfAgGg+D-EAvRgZAHCcFwQD5LowAAawkO4GzAZ5kEechcN6MAsMcZx-xSZxuBgKCumEPp7HonD-wAKwwAAPAAhAjiJgO48TgCMUjgHjGIA8gYHeJSACZ5NwhtuBY5AQC47hCN6aMoDQ94nlgQs+HmZAAA5bRAMyIGcDguneXSzMwAJ-DQsQAEdyBoMR2PsTjuKQbCFOcPoaEo7hqP-SwpGsABFcgIHgDT-xgew+Us6ykFUrKuisZ53X6LjdKgaA5P-dpUBy+CIrgIcgA))
- [x] Fix the new lint errors
- [ ] Use [`prettylint`](https://github.com/ikatyang/prettylint) for :sparkles:

### CODE REVIEW

**Comment by iAmWillShepherd** [comment]
I recommend you get the input from @desktop/design since they are most often the people working in these files. Prettier's opinions may not agree with theirs. Also, they haven't mentioned styling being an issue for them, so this PR may not be necessary.

**Comment by j-f1** [comment]
I’ve changed the SCSS quote style back to double quotes, since that’s what the SCSS used before. @desktop/design Should I change the quotes to single quotes for consistency with the TS?

File: docs/process/testing.md [hunk 1]
```diff
@@ -1,366 +1,491 @@
...
+        https://central.github.com/deployments/desktop/desktop/latest/darwin
+    * [ ] Homebrew package manager: $ brew cask install github-desktop
+  * [ ] Windows:
+        https://central.github.com/deployments/desktop/desktop/latest/win32
+    * [ ] Chocolatey package manager: c:\> choco install github-desktop
+    * [ ] 64-bit and up
+  * [ ] Data is retained if you download and open a fresh copy
+* [ ] Release notes page is up-to-date
+* [ ] Help page is accessible
+* [ ] 'Please update' notifcation shown in Classic apps
```
Comment (j-f1) [inline docs/process/testing.md]: @tierninho Are these changes compatible with Quality, or should I ignore this file?

**Comment by j-f1** [comment]
@shiftkey do you have any ideas about why the AppVeyor build is failing?

**Comment by j-f1** [comment]
💄

@iAmWillShepherd Can you request 👀 from @desktop/design?

**Comment by timofonic** [comment]
@j-f1 Nice!

@iAmWillShepherd Any news from design team? Are those  @donokuda and @niik?

**Comment by j-f1** [comment]
@timofonic. It’s the weekend; someone will probably have a look at it on Monday.

File: package.json [hunk 2]
```diff
@@ -5,32 +5,46 @@
...
-    "test": "yarn test:unit && yarn test:integration",
+    "cli":
+      "ts-node --require ./app/test/globals.ts --require ./app/src/cli/dev-commands-global.js app/src/cli/main.ts",
+    "test:integration":
+      "cross-env TEST_ENV=1 ELECTRON_NO_ATTACH_CONSOLE=1 xvfb-maybe mocha -t 30000 --require ts-node/register app/test/integration/*.ts",
+    "test:unit":
+      "cross-env GIT_AUTHOR_NAME=\"Joe Bloggs\" GIT_AUTHOR_EMAIL=\"joe.bloggs@somewhere.com\" GIT_COMMITTER_NAME=\"Joe Bloggs\" GIT_COMMITTER_EMAIL=\"joe.bloggs@somewhere.com\" TEST_ENV=1 ELECTRON_NO_ATTACH_CONSOLE=1 xvfb-maybe electron-mocha -t 10000 --renderer --require ts-node/register --require ./app/test/globals.ts app/test/unit/*.{ts,tsx} app/test/unit/**/*.{ts,tsx}",
+    "test":
+      "cross-env FORCE_COLOR=1 run-p --aggregate-output test:unit test:integration",
```
Comment (shiftkey) [inline package.json]: This seems unrelated to the current work, and is broken on Travis:

```
$ cross-env FORCE_COLOR=1 run-p --aggregate-output test:unit test:integration
xvfb-run:
   error:
   Xvfb
   failed
   to
   start
Failed with exit code: 1
```

Comment (j-f1) [reply package.json]: Should I revert it 🔄?

File: package.json [hunk 3]
```diff
@@ -5,32 +5,46 @@
...
+    "lint:fix": "cross-env FORCE_COLOR=1 run-p --aggregate-output lint:*:fix",
+    "lint:prettier": "cross-env FORCE_COLOR=1 run-p prettier:*",
+    "lint:prettier:fix": "cross-env FORCE_COLOR=1 run-p 'prettier:* --write'",
+    "prettier:md": "prettier --list-different '**/*.md'",
+    "prettier:scss": "prettier --list-different '**/*.scss'",
+    "prettier:json": "prettier --list-different '**/*.json'",
+    "lint:src": "yarn tslint && yarn eslint-check && yarn eslint",
+    "lint:src:fix": "yarn tslint --fix && yarn eslint --fix",
+    "tslint":
+      "cross-env NODE_OPTIONS='-r ts-node/register' tslint ./{script,tslint-rules}/*.ts ./app/{src,typings,test}/**/*.{ts,tsx}",
```
Comment (shiftkey) [inline package.json]: It's not clear to me why we need `NODE_OPTIONS` here when we're spawning `tslint` - that knows how to parse TS files.

Comment (j-f1) [reply package.json]: `NODE_OPTIONS` is to let us avoid having to transpile our TSLint rules before using them.

File: package.json [hunk 4]
```diff
@@ -5,32 +5,46 @@
   },
   "description": "GitHub Desktop build dependencies",
   "scripts": {
-    "cli": "ts-node --require ./app/test/globals.ts --require ./app/src/cli/dev-commands-global.js app/src/cli/main.ts",
-    "test:integration": "cross-env TEST_ENV=1 ELECTRON_NO_ATTACH_CONSOLE=1 xvfb-maybe mocha -t 30000 --require ts-node/register app/test/integration/*.ts",
-    "test:unit": "cross-env GIT_AUTHOR_NAME=\"Joe Bloggs\" GIT_AUTHOR_EMAIL=\"joe.bloggs@somewhere.com\" GIT_COMMITTER_NAME=\"Joe Bloggs\" GIT_COMMITTER_EMAIL=\"joe.bloggs@somewhere.com\" TEST_ENV=1 ELECTRON_NO_ATTACH_CONSOLE=1 xvfb-maybe electron-mocha -t 10000 --renderer --require ts-node/register --require ./app/test/globals.ts app/test/unit/*.{ts,tsx} app/test/unit/**/*.{ts,tsx}",
-    "test": "yarn test:unit && yarn test:integration",
+    "cli":
```
Comment (shiftkey) [inline package.json]: Can you also undo the formatting changes here (I guess this is VSCode's doing) so it's easier to see what's actually changed in this file?

Comment (j-f1) [reply package.json]: These changes are intentionally part of this PR — I’ve enabled Prettier for the JSON files.

Comment (j-f1) [reply package.json]: The diff for the commit (https://github.com/desktop/desktop/pull/3569/commits/88002e02b854df1cd0684c5228ca66349313cc9a) shows the semantic changes in that PR fairly well.

File: package.json (hunk 3 above)
Comment (shiftkey) [reply package.json]: The whole task takes ~1.5s on my machine, and I couldn't really spot a difference between the three different ways to invoke it. This feels like a bit of yak shaving IMO.

```shellsession
$ time yarn tslint
yarn run v1.3.2
$ cross-env NODE_OPTIONS='-r ts-node/register' tslint ./{script,tslint-rules}/*.ts ./app/{src,typings,test}/**/*.{ts,tsx}
✨  Done in 1.51s.
yarn tslint  2.39s user 0.19s system 142% cpu 1.808 total

$ time yarn tslint
yarn run v1.3.2
$ cross-env tslint ./{script,tslint-rules}/*.ts ./app/{src,typings,test}/**/*.{ts,tsx}
✨  Done in 1.46s.
yarn tslint  2.43s user 0.17s system 147% cpu 1.768 total

$ time yarn tslint
yarn run v1.3.2
$ tslint ./{script,tslint-rules}/*.ts ./app/{src,typings,test}/**/*.{ts,tsx}
✨  Done in 1.35s.
yarn tslint  2.32s user 0.15s system 150% cpu 1.648 total
```

File: package.json (hunk 4 above)
Comment (shiftkey) [reply package.json]: > I’ve enabled Prettier for the JSON files.

Could we not do this right now? We only use JSON files in a few mission-critical places, and you're currently introducing a lot of churn in here on top of it which is making this a challenge to review.

File: package.json [hunk 5]
```diff
@@ -128,6 +142,7 @@
     "electron-builder": "19.48.3",
     "electron-mocha": "^5.0.0",
     "electron-packager": "^10.1.0",
-    "electron-winstaller": "2.5.2"
+    "electron-winstaller": "2.5.2",
+    "npm-run-all": "^4.1.2"
```
Comment (shiftkey) [inline package.json]: This should be a proper `dependency` - the `electron-*` packages are only here because one of them (probably `electron-builder) demands to be a `devDependency`.

File: package.json (hunk 2 above)
Comment (shiftkey) [reply package.json]: Yes, let's run them sequentially for now and then revisit later.

File: package.json [hunk 6]
```diff
@@ -5,32 +5,46 @@
...
+      "ts-node --require ./app/test/globals.ts --require ./app/src/cli/dev-commands-global.js app/src/cli/main.ts",
+    "test:integration":
+      "cross-env TEST_ENV=1 ELECTRON_NO_ATTACH_CONSOLE=1 xvfb-maybe mocha -t 30000 --require ts-node/register app/test/integration/*.ts",
+    "test:unit":
+      "cross-env GIT_AUTHOR_NAME=\"Joe Bloggs\" GIT_AUTHOR_EMAIL=\"joe.bloggs@somewhere.com\" GIT_COMMITTER_NAME=\"Joe Bloggs\" GIT_COMMITTER_EMAIL=\"joe.bloggs@somewhere.com\" TEST_ENV=1 ELECTRON_NO_ATTACH_CONSOLE=1 xvfb-maybe electron-mocha -t 10000 --renderer --require ts-node/register --require ./app/test/globals.ts app/test/unit/*.{ts,tsx} app/test/unit/**/*.{ts,tsx}",
+    "test":
+      "cross-env FORCE_COLOR=1 run-p --aggregate-output test:unit test:integration",
     "test:setup": "ts-node script/test-setup.ts",
     "test:review": "ts-node script/test-review.ts",
-    "postinstall": "cd app && yarn install --force && cd .. && git submodule update --recursive --init && yarn compile:tslint",
```
Comment (shiftkey) [inline package.json]: It looks like dropping `yarn compile:tslint` here affects the IDE integration with TSLint. Without the rules being transpiled to JS, you'll probably see an error like this:

**Comment by iAmWillShepherd** [comment]
@j-f1 the design team has been busy with internal stuff and since holidays are upon us, they are unlikely to get to this - that goes for me too - until next year.

**Comment by j-f1** [comment]
@shiftkey I think those reverts cover the issues you brought up.

@iAmWillShepherd
> the design team has been busy with internal stuff and since holidays are upon us, they are unlikely to get to this - that goes for me too - until next year

👍 That’s fine. Feel free to wait :)

**Comment by donokuda** [comment]
I'm 👍 for bringing these changes in. I'm also impartial to our coding style as long as we're all consistent in the end.

How are these styles enforced? Is it editor specific? Do we lint on CI as well?

**Comment by j-f1** [comment]
@donokuda 

> How are these styles enforced?

Prettier supports [all kinds of languages](https://prettier.io/), including CSS, Markdown, and GraphQL (for when we upgrade to API v4).

> Is it editor specific?

Nope, just make sure your editor has [a Prettier plugin](https://prettier.io/docs/en/editors.html) installed for the best experience. Otherwise, you can run `yarn lint:prettier` or `yarn lint` to run Prettier.

> Do we lint on CI as well?

Yup 👍 ✅

File: package.json [hunk 7]
```diff
@@ -24,12 +24,13 @@
     "clean:tslint": "rimraf tslint-rules/*.js",
     "compile:tslint": "tsc -p tslint-rules",
     "compile:script": "tsc -p script",
-    "lint": "yarn compile:tslint && yarn compile:script && yarn tslint && yarn eslint-check && yarn eslint",
-    "lint:fix": "yarn compile:tslint && yarn tslint:fix && yarn eslint:fix",
+    "lint": "yarn lint:src && yarn lint:prettier",
+    "lint:fix": "yarn lint:src:fix && yarn lint:prettier --write",
+    "lint:prettier": "prettier '**/*.md' '**/*.scss' --list-different",
```
Comment (shiftkey) [inline package.json]: 💭  while this command here is nice and succinct about what needs to be fixed, someone first encountering this for the first time doesn't really get much context:

```shellsession
~/src/desktop/ yarn lint
yarn run v1.3.2
$ yarn lint:src && yarn lint:prettier
$ yarn tslint && yarn eslint-check && yarn eslint
$ tslint ./{script,tslint-rules}/*.ts ./app/{src,typings,test}/**/*.{ts,tsx}
$ eslint --print-config .eslintrc.* | eslint-config-prettier-check
No rules that are unnecessary or conflict with Prettier were found.
$ ts-node script/eslint.ts

$ prettier '**/*.md' '**/*.scss' --list-different
CONTRIBUTING.md
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

We often have to go in and explain what's happening here when a PR is opened and fails CI, but maybe we can do better to reduce that work:

 - some documentation around "things you should do before submitting a PR"
 - wrapping this in a custom script to handle the exit code on error, display some context around this output and indicate what should be done next

Or maybe a combination of the two?

Comment (j-f1) [reply package.json]: 👍 I’ll look into making putting a message if it failed.

**Comment by j-f1** [comment]
[review-plz]

**Comment by shiftkey** [comment]
> ❔ Lint the JS/TS in Markdown (`eslint-plugin-markdown`) (` ```ts ` blocks aren’t supported yet) (blocked on https://github.com/eslint/eslint-plugin-markdown/issues/82)

This seems like a nice-to-have rather than something we should hold up the PR for. I'm fine with tracking this in a separate issue if the upstream issue isn't going to be resolve soon.

**Comment by j-f1** [comment]
I’ve tested the above commit with `true && true || (true && false)` (parens are necessary) in Bash, and it seems to work

**Comment by shiftkey** [comment]
> I’ve tested the above commit with `true` and `false` in Bash, and it seems to work

Note to self: test this in PowerShell and `cmd` and confirm it also works as expected

File: app/styles/ui/_button.scss [hunk 8]
```diff
@@ -26,14 +26,16 @@
     box-shadow: 0 0 0 1px var(--secondary-button-focus-shadow-color);
   }
 
-  &:disabled { opacity: 0.6; }
+  &:disabled {
+    opacity: 0.6;
+  }
```
[the remark this answers is not in the document]
Comment (j-f1) [reply app/styles/ui/_button.scss]: > Would a PR be accepted for making single-declaration rules like `foo.bar { baz: 'quux'; }` stay on one line be accepted?

> I doubt that. Prettier tries to have as few options as possible.

File: package.json [hunk 9]
```diff
@@ -24,12 +24,13 @@
     "clean:tslint": "rimraf tslint-rules/*.js",
     "compile:tslint": "tsc -p tslint-rules",
     "compile:script": "tsc -p script",
-    "lint": "yarn compile:tslint && yarn compile:script && yarn tslint && yarn eslint-check && yarn eslint",
-    "lint:fix": "yarn compile:tslint && yarn tslint:fix && yarn eslint:fix",
+    "lint": "yarn lint:src && yarn lint:prettier || echo '\u001b[1m\u001b[32m→ To fix some of these errors, run \u001b[39m\u001b[22m\u001b[1m\u001b[32m\u001b[4myarn eslint:fix\u001b[24m\u001b[39m\u001b[22m'",
```
[the remark this answers is not in the document]
Comment (shiftkey) [reply package.json]: One more mention of `eslint:fix` that needs to be updated: https://github.com/desktop/desktop/blob/master/script/eslint.ts#L32

File: package.json [hunk 10]
```diff
@@ -24,12 +24,13 @@
     "clean:tslint": "rimraf tslint-rules/*.js",
     "compile:tslint": "tsc -p tslint-rules",
     "compile:script": "tsc -p script",
-    "lint": "yarn compile:tslint && yarn compile:script && yarn tslint && yarn eslint-check && yarn eslint",
-    "lint:fix": "yarn compile:tslint && yarn tslint:fix && yarn eslint:fix",
+    "lint": "yarn lint:src && yarn lint:prettier || (echo '\u001b[1m\u001b[32m→ To fix some of these errors, run \u001b[39m\u001b[22m\u001b[1m\u001b[32m\u001b[4myarn lint:fix\u001b[24m\u001b[39m\u001b[22m' && false)",
+    "lint:fix": "yarn lint:src:fix && yarn lint:prettier --write",
+    "lint:prettier": "prettier '**/*.md' '**/*.scss' --list-different",
```
Comment (shiftkey) [inline package.json]: Drop the quote marks from these paths so they get expanded by the shell. Just tested this works in Git Bash, Git CMD and PowerShell (all the Windows shells I can think of, basically):

Comment (j-f1) [reply package.json]: Does `**/*.{md,scss}` work, too?

File: package.json [hunk 11]
```diff
@@ -24,12 +24,13 @@
     "clean:tslint": "rimraf tslint-rules/*.js",
     "compile:tslint": "tsc -p tslint-rules",
     "compile:script": "tsc -p script",
-    "lint": "yarn compile:tslint && yarn compile:script && yarn tslint && yarn eslint-check && yarn eslint",
-    "lint:fix": "yarn compile:tslint && yarn tslint:fix && yarn eslint:fix",
+    "lint": "yarn lint:src && yarn lint:prettier || (echo '\u001b[1m\u001b[32m→ To fix some of these errors, run \u001b[39m\u001b[22m\u001b[1m\u001b[32m\u001b[4myarn lint:fix\u001b[24m\u001b[39m\u001b[22m' && false)",
```
Comment (shiftkey) [inline package.json]: The use of `false` here doesn't work for non-bash shells. This affects both CMD and PowerShell on Windows:

```shellsession
> yarn lint
yarn run v1.3.2
$ yarn lint:src && yarn lint:prettier || (echo '→ To fix some of these errors, run yarn lint:fix' && false)
$ yarn tslint && yarn eslint-check && yarn eslint
$ tslint ./{script,tslint-rules}/*.ts ./app/{src,typings,test}/**/*.{ts,tsx}
$ eslint --print-config .eslintrc.* | eslint-config-prettier-check
No rules that are unnecessary or conflict with Prettier were found.
$ ts-node script/eslint.ts

$ prettier **/*.md **/*.scss --list-different
.github\ISSUE_TEMPLATE.md
docs\contributing\linux-testing-and-you.md
docs\technical\syntax-highlighting.md
app\styles\ui\_repository-list.scss
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
'→ To fix some of these errors, run yarn lint:fix'
'false' is not recognized as an internal or external command,
operable program or batch file.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I think we might have to use the `prettier` API here if we want better control of the output and exit codes across multiple shells. I'll take some time today to see if I can PR something into your branch that's similar to `script/eslint.ts`.

File: package.json (hunk 10 above)
Comment (shiftkey) [reply package.json]: Seems fine here:

File: package.json (hunk 11 above)
Comment (shiftkey) [reply package.json]: See https://github.com/j-f1/forked-desktop/pull/1 for a proposed solution

File: .travis.yml [hunk 12]
```diff
@@ -57,10 +57,7 @@ install:
   - yarn install --force
 
 script:
-  - yarn lint
-  - yarn build:prod
-  - yarn test:setup
-  - yarn test
+  - yarn lint && yarn build:prod && yarn test:setup && yarn test
```
Comment (shiftkey) [inline .travis.yml]: Unlike the other CI platforms, Travis happily marched on after it encountered a linting issue: https://travis-ci.org/desktop/desktop/builds/326538605#L717

This is the documented way to make the build fail fast: https://docs.travis-ci.com/user/customizing-the-build/#Customizing-the-Build-Step

**Comment by j-f1** [comment]
Hey @desktop/design, would you prefer single or double quotes in the SCSS?

**Comment by niik** [comment]
> Hey @desktop/design, would you prefer single or double quotes in the SCSS?

I can't say that I have a preference. What's the default in prettier?

**Comment by j-f1** [comment]
> What's the default in prettier?

The default is a double quote, but we’ve set the default to single quotes for the TypeScript.

**Comment by donokuda** [comment]
I'm fine with using single quotes to be consistent with the rest of the project.

**Comment by shiftkey** [comment]
There's a shiny new `prettier` error here that the script didn't uncover:
 


b562b63 should make this clearer while I figure out the underlying issue

**Comment by shiftkey** [comment]
@iAmWillShepherd the earlier merge abf3fdf introduced the issue and aa8d61f7 bumped it further to fix it

**Comment by iAmWillShepherd** [comment]
@shiftkey yes, I see it now. Not sure why I didn't see that second commit, aa8d61f7, though.

**Comment by j-f1** [comment]
https://github.com/prettier/prettier/issues/3742

**Comment by j-f1** [comment]
🚰

File: .prettierignore [hunk 13]
```diff
@@ -1,2 +1,15 @@
...
+.DS_Store
+.awcache
+.idea/
+.eslintcache
+
+app/static/common
+app/test/fixtures
+gemoji
 *.json
-*.md
```
Comment (iAmWillShepherd) [inline .prettierignore]: I'm pretty sure we don't want prettier messing with our md files.

Comment (j-f1) [reply .prettierignore]: Why not?

Comment (iAmWillShepherd) [reply .prettierignore]: Because it adds unneeded friction. Our build shouldn't fail because something unrelated to building a project fails. An example is me adding something to our technical docs causing CI to fail.

Comment (j-f1) [reply .prettierignore]: Coding style is unrelated to building the project, but we still enforce it 😄

This PR is just expanding our “style guide” (provided by Prettier) to the Markdown and SCSS files.

Comment (iAmWillShepherd) [reply .prettierignore]: So my question is if my MD style doesn't agree with Prettier, will CI fail and prevent me from merging? If the answer to that is yes, then this needs to be removed. We use these files internally and haven't had any issues with a lack of prettier's styling.

To state another way, we have had no issues with styling markdown.

Comment (shiftkey) [reply .prettierignore]: I'm in agreement with @iAmWillShepherd for a couple of reasons:

 - the output artefacts that are mission-critical are the code and stylesheets. Documentation should be easy to write, without requiring the same strictness
 - if all that Prettier is doing is enforcing a consistent line length, that shouldn't block CI
 - help ensure IDEs format Markdown files well when editing is more valuable

Comment (j-f1) [reply .prettierignore]: > if all that Prettier is doing is enforcing a consistent line length, that shouldn't block CI

It also removes unnecessary linebreaks and formats code blocks and enforces consistent styles for lists, emphasis, headers, and several other things.

If you feel that requiring a consistent line length isn’t helpful, there’s [an option](https://prettier.io/docs/en/options.html#prose-wrap) to disable wrapping.

Comment (shiftkey) [reply .prettierignore]: @j-f1 the line length thing is just the one I picked out.

I'm fine with using prettier to format Markdown when editing, but enforcing it on CI adds a barrier to contribution that I feel is unnecessary.

Comment (j-f1) [reply .prettierignore]: How about adding a new npm script to prettify the Markdown, but remove the check from the CI build (or list the non-pretty files, but not fail the build)?

Comment (shiftkey) [reply .prettierignore]: I think that works. @iAmWillShepherd?

Comment (iAmWillShepherd) [reply .prettierignore]: That's fine, but I don't find having a script for this to be useful. I don't think we've had any issues with our markdown files to date and I have a hard time caring about consistent styling for these files. I am, however, willing to defer this one so long as this won't block me from merging code that doesn't meet prettier's standards.

**Comment by iAmWillShepherd** [comment]
Looks like it failing on a markdown file.
